lib/dns/cache_test.go

```diff
@@ -215,8 +215,8 @@ func TestConcurrency(t *testing.T) {
 
 	wg.Wait()
 
-	// We need to have an empty cache now
-	if c.Count() != 0 {
-		t.Fatalf("Failed to remove eveything")
+	// We should remove at least one item from the cache
+	if c.Count() == samplesize {
```

Comment:
So are we just testing that we don't panic here? Without any specific thing to test for at the end, not sure why we have this test.